wisdem/precomp/precomp_to_beamdyn.py

```diff
@@ -0,0 +1,141 @@
+import numpy as np
+
+# --------------------------------------------------------------------------------}
+# --- Tools to roto-translate 6x6 stiffness and mass matrix
+# --------------------------------------------------------------------------------{
+class TransformCrossSectionMatrix(object):
+
+    def CrossSectionTranslationMatrix(self, x, y):
+        T = np.eye(6)
+        T[0,5] = y
+        T[1,5] = -x
+        T[2,3] = -y
+        T[2,4] = x
+        return T
+
+    def CrossSectionRotationMatrix(self, alpha):
+        c=np.cos(alpha)
+        s=np.sin(alpha)
+        R1=[[c,s,0],
+            [-s,c,0],
+            [0,0,1]]
+        R=np.vstack((np.hstack((R1, np.zeros((3,3)))),
+           np.hstack((np.zeros((3,3)), R1))))
+        return R
+
+    def CrossSectionRotoTranslationMatrix(self, M1, x, y, alpha):
+        # Translation
+        T = self.CrossSectionTranslationMatrix(x, y)
+        M2 = T.T @ M1 @ T 
+        # Rotation 
+        R = self.CrossSectionRotationMatrix(alpha)
+        M3 = R @ M2 @ R.T
```

Comment:
@ptrbortolotti Could you check if this should be `R.T @ M2 @ R` instead? Depending on how R is formed and how alpha direction is defined, it can be RAR.T or R.TAR. I think for here, it may be `R.T @ M2 @ R`.

Reply:
I don't think so?

![I_matrices](https://github.com/user-attachments/assets/beaa7272-1b94-4402-83ec-783d7c916415)

we can also run an extra test by generating an I matrix for a rectangular cross section and then rototranslate it, and see what happens. I haven't done that check

Reply:
ok some extra bit of checks @yqliaohk. `R.T @ M2 @ R` works just fine if I flip the sign in `alpha`, which is defined here. nrel.gov right now is down and I can't access the PreComp user manual, but that will be my next check
```python
    I_ref = transform.CrossSectionRotoTranslationMatrix(
        I_cg, 
        -x_cg, 
        -y_cg, 
        - (Tw_iner - aero_twist),
        )

Reply:
Thank you for doing the checks! I think I might know where the discrepancy comes from. The reason that a flipped sign is currently needed to work with `R.T @ M2 @ R` is that `- (Tw_iner - aero_twist)` is how precomp actually computes the angle between the inertia frame and the reference frame [https://github.com/WISDEM/WISDEM/blob/14ad23788496bc130d69d613b9c408f9615f21b7/wisdem/precomp/properties.py#L636](url)